apps/desktop/src/routes/[projectId]/+layout.svelte

```diff
@@ -164,6 +165,22 @@
 		}
 	});
 
+	// TODO(mattias): This is an ugly hack, fix it somehow?
+	// I want to flush project metrics to local storage before e.g. switching
+	// to a different project. Since `projectMetrics` is defined in layout.ts
+	// we get no heads up when it is about to change, and reactively updated
+	// in this scope through `LayoutData`. Even at time of unMount in e.g.
+	// metrics reporter it seems as if the projectMetrics variable is already
+	// referencing the new instance.
+	let lastProjectMetrics: ProjectMetrics | undefined;
+	$effect(() => {
+		if (lastProjectMetrics) {
+			lastProjectMetrics.saveToLocalStorage();
+		}
+		lastProjectMetrics = projectMetrics;
+		projectMetrics.loadFromLocalStorage();
+	});
```

Comment:
@Caleb-T-Owens this could go away if we move stuff from layout.ts into layout.svelte, but I want to land it now. Can I get your blessing? 😅